### PR TITLE
Adding evaluation of TURNA-encoder

### PR DESCRIPTION
**Pull request by zeynepyirmibes**
In this PR, the evaluation of fine-tuned TURNA-encoder models was implemented. Changes include:

- Fix of the [test_params issue](https://github.com/boun-tabi-LMG/turkish-lm-tuner/issues/70) in the evaluation script. 
- Fix of erroneous label post-processing of classification and STS datasets. 
- Evaluation of fine-tuned TURNA-encoder models.

### CODE REVIEW

File: turkish_lm_tuner/t5_classifier.py [hunk 1]
```diff
@@ -18,7 +21,12 @@ class T5ForClassification(T5PreTrainedModel):   # nn.Module
     def __init__(self, pretrained_model_name, config, num_labels, problem_type, dropout_prob=0.1):
         super().__init__(config)
 
-        self.encoder = T5EncoderModel.from_pretrained(pretrained_model_name)
+        try:
+            self.encoder = T5EncoderModel.from_pretrained(pretrained_model_name)
+        except Exception as e:
+            pretrained_model_name = config._name_or_path
+            self.encoder = T5EncoderModel.from_pretrained(pretrained_model_name)
+
```
Comment (gokceuludogan) [inline turkish_lm_tuner/t5_classifier.py:29]: Do we really need exception handling here? When does the encoder initialization fail with `pretrained_model_name`?

File: turkish_lm_tuner/tr_datasets.py [hunk 2]
```diff
@@ -156,7 +156,11 @@ def convert_sts_label(label):
             try:
                 return(float(label.strip()))
             except:
-                return 0
+                try:
+                    return(float(label))
+                except:
+                    return 0
+
```
Comment (gokceuludogan) [inline turkish_lm_tuner/tr_datasets.py:163]: Doesn't `float(label.strip())` cover `float(label)` already?

Did we check the outputs to see if they contain similarity scores where they cannot immediately cast to float?

**Review by gokceuludogan — APPROVED** [review]
Thanks a lot for your efforts on fixing the TURNA encoder and its evaluation! 

I've left a few minor comments, but other than that, I'm happy to approve this PR!

File: turkish_lm_tuner/t5_classifier.py (hunk 1 above)
Comment (zeynepyirmibes) [reply turkish_lm_tuner/t5_classifier.py:29]: This is really bad code I realize :D but there was a problem with the `from_pretrained` method, when it calls the init of T5forSequenceClassification, that argument is overwritten as the config.. I couldn't solve it, so I added this.. If I can find a solution, I will fix it! 👍

File: turkish_lm_tuner/tr_datasets.py (hunk 2 above)
Comment (zeynepyirmibes) [reply turkish_lm_tuner/tr_datasets.py:163]: When we use conditional generation, the output is a string, so `float(label.strip())` works. When we use classification, the output is a number, so the `strip()` function naturally gives an error, so I try converting into a float.

File: turkish_lm_tuner/t5_classifier.py (hunk 1 above)
Comment (gokceuludogan) [reply turkish_lm_tuner/t5_classifier.py:29]: Can we predict if it's going to fail beforehand and update the pretrained_model_name accordingly? In what scenarios do we use the first from_pretrained method compared to the other one?

File: turkish_lm_tuner/tr_datasets.py (hunk 2 above)
Comment (gokceuludogan) [reply turkish_lm_tuner/tr_datasets.py:163]: I see. Rather than handling it with exceptions, I suggest checking its type to choose the conversion. It would be much clearer.
